Source/Diagnostics/ReducedDiags/ParticleExtrema.cpp

```diff
@@ -0,0 +1,459 @@
+
+        // gmin
+        Real gmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            Real ux = p.rdata(PIdx::ux);
+            Real uy = p.rdata(PIdx::uy);
+            Real uz = p.rdata(PIdx::uz);
+            Real us = ux*ux + uy*uy + uz*uz;
+            return std::sqrt(1.0 + us*inv_c2);
+        });
+        ParallelDescriptor::ReduceRealMin(gmin);
+
+        // gmax
+        Real gmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            Real ux = p.rdata(PIdx::ux);
+            Real uy = p.rdata(PIdx::uy);
+            Real uz = p.rdata(PIdx::uz);
+            Real us = ux*ux + uy*uy + uz*uz;
+            return std::sqrt(1.0 + us*inv_c2);
+        });
+        ParallelDescriptor::ReduceRealMax(gmax);
+
+        // wmin
+        Real wmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::w); });
+        ParallelDescriptor::ReduceRealMin(wmin);
+
+        // wmax
+        Real wmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::w); });
+        ParallelDescriptor::ReduceRealMax(wmax);
+
+#if (defined WARPX_QED)
+        // compute chimin and chimax
+        Real chimin = 0.0_rt;
+        Real chimax = 0.0_rt;
+        GetExternalEField get_externalE;
+        GetExternalBField get_externalB;
+
+        if (myspc.has_breit_wheeler() || myspc.has_quantum_sync())
+        {
+            chimin = ReduceMin( myspc,
+            [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+            {
+#if (defined WARPX_DIM_RZ)
+                ParticleReal xp = p.pos(0)*std::cos(p.rdata(PIdx::theta));
+                ParticleReal yp = p.pos(0)*std::sin(p.rdata(PIdx::theta));
+#elif (defined WARPX_DIM_XZ)
+                ParticleReal xp = p.pos(0);
+                ParticleReal yp = 0.0_rt;
+#else
+                ParticleReal xp = p.pos(0);
+                ParticleReal yp = p.pos(1);
+#endif
+                ParticleReal zp = p.pos(index_z);
+
+                ParticleReal ex = 0._rt, ey = 0._rt, ez = 0._rt;
+                ParticleReal bx = 0._rt, by = 0._rt, bz = 0._rt;
+
+                Array4<const amrex::Real> ex_arr;
+                Array4<const amrex::Real> ey_arr;
+                Array4<const amrex::Real> ez_arr;
+                Array4<const amrex::Real> bx_arr;
+                Array4<const amrex::Real> by_arr;
+                Array4<const amrex::Real> bz_arr;
+                IndexType ex_type, ey_type, ez_type;
+                IndexType bx_type, by_type, bz_type;
+                GpuArray<amrex::Real, 3> dx_arr, xyzmin_arr;
+                Dim3 lo;
+
+                int n_rz_azimuthal_modes;
+                int nox;
+                bool galerkin_interpolation;
+
+                get_externalE(i, ex, ey, ez);
+                get_externalB(i, bx, by, bz);
```

Comment:
@atmyers @lucafedeli88 
How can I get `i` in this `ReduceMax/Min` loop for `get_externalE(i, ex, ey, ez)` etc.
Something like?
```
chimin = ReduceMin( myspc,
            [=] AMREX_GPU_HOST_DEVICE (const PType& p, int i) -> Real
            { ... }
```